### PR TITLE
SAM4L: Add DAC driver

### PR DESCRIPTION
**Pull request by bradjc**
- Pretty basic DAC support.
- Userland test app.

### CODE REVIEW

File: capsules/src/dac.rs [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+//! -----
+//!
+//! ```rust
+//! let dac = static_init!(
+//!     capsules::dac::Dac<'static>,
+//!     capsules::dac::Dac::new(&mut sam4l::dac::DAC));
+//! ```
+
+use kernel::{AppId, Driver};
+use kernel::ReturnCode;
```
Comment (ppannuto) [inline capsules/src/dac.rs]: 🚲 : any reason this isn't just `use kernel::{AppId, Driver, ReturnCode};` ?

File: kernel/src/hil/dac.rs [hunk 2]
```diff
@@ -0,0 +1,12 @@
+//! Interface for digital to analog converters.
+
+use returncode::ReturnCode;
+
+/// Simple interface for using the DA.
```
Comment (ppannuto) [inline kernel/src/hil/dac.rs]: `DA` -> `DAC`

File: capsules/src/dac.rs (hunk 1 above)
Comment (brghena) [inline capsules/src/dac.rs]: I really wish rustfmt would handle this for us...

**Comment by ppannuto** [comment]
We are fairly behind on rustfmt version (sigh..) and they are responsive to
feature requests.

It's possible that it already does / could

File: capsules/src/dac.rs (hunk 1 above)
Comment (brghena) [inline capsules/src/dac.rs]: Looks like they haven't gotten there yet though. https://github.com/rust-lang-nursery/rustfmt/issues/1383

**Comment by alevy** [comment]
@ppannuto if/when we upgrade rustfmt, it's gonna probably be changes to every file (they changed the style significantly after a long RFC process of stabilizing it). So we should just do it when we're not doing other stuff :)

**Comment by alevy** [comment]
@hsiehju looks like you had it right all along!

**Comment by bradjc** [comment]
Updated.

And re: rustfmt, I say we wait till 1.0. Other than the 100 byte line limit, it's working pretty well.

**Pull request merged**